src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -1841,6 +1841,20 @@ public Double zscore(final byte[] key, final byte[] member) {
     return (score != null ? new Double(score) : null);
   }
 
+  public Tuple zpopmax(final byte[] key) {
+    checkIsInMultiOrPipeline();
+    client.zpopmax(key);
+    
+    List<byte[]> l = client.getBinaryMultiBulkReply();
+    return l.isEmpty() ? null : new Tuple(l.get(0), BuilderFactory.DOUBLE.build(l.get(1)));
```

Comment:
You've created `BuilderFactory.TUPLE`. Can't we use it here to avoid code repetition?

Reply:
Done. With the mention that I changed the null check in `BuilderFactory.TUPLE` to look for empty casted List parameters instead of a null parameter. For now, it only receives non-null values.